### PR TITLE
Fix #92: Remove File.Type

### LINKED ISSUE #92: Checking that a .gz file is empty fails with an exception
If I try to check that some gzipped file is empty (irrespectively of whether the original file was empty or not), it fails to [create a `BufferedSource`](https://github.com/pathikrit/better-files/blob/v2.16.0/core/src/main/scala/better/files/File.scala#L879-L881) and throws an exception:

```
scala> file"foo.txt.gz".isEmpty
java.nio.charset.MalformedInputException: Input length = 1
  at java.nio.charset.CoderResult.throwException(CoderResult.java:281)
  at sun.nio.cs.StreamDecoder.implRead(StreamDecoder.java:339)
  at sun.nio.cs.StreamDecoder.read(StreamDecoder.java:178)
  at java.io.InputStreamReader.read(InputStreamReader.java:184)
  at java.io.BufferedReader.fill(BufferedReader.java:161)
  at java.io.BufferedReader.read(BufferedReader.java:182)
  at scala.io.BufferedSource$$anonfun$iter$1$$anonfun$apply$mcI$sp$1.apply$mcI$sp(BufferedSource.scala:40)
  at scala.io.Codec.wrap(Codec.scala:69)
  at scala.io.BufferedSource$$anonfun$iter$1.apply(BufferedSource.scala:40)
  at scala.io.BufferedSource$$anonfun$iter$1.apply(BufferedSource.scala:40)
  at scala.collection.Iterator$$anon$9.next(Iterator.scala:162)
  at scala.collection.Iterator$$anon$16.hasNext(Iterator.scala:598)
  at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:408)
  at scala.io.Source.hasNext(Source.scala:238)
  at scala.collection.Iterator$class.isEmpty(Iterator.scala:330)
  at scala.io.Source.isEmpty(Source.scala:190)
  at better.files.File.isEmpty(File.scala:378)
  ... 42 elided
```

### PR DESCRIPTION
**Pull request by pathikrit**

### CODE REVIEW

File: core/src/main/scala/better/files/File.scala [hunk 1]
```diff
@@ -722,24 +722,31 @@ class File private(val path: Path) {
...
+    } else if (isRegularFile(linkOptions))  {
+      toJava.length() == 0
+    } else {
+      notExists(linkOptions)
     }
   }
 
+  def isNotEmpty(implicit linkOptions: File.LinkOptions = File.LinkOptions.default): Boolean = {
+    !isEmpty(linkOptions)
+  }
```
Comment (laughedelic) [inline core/src/main/scala/better/files/File.scala]: If this is a new method in the upcoming release, consider renaming it to `nonEmpty` by analogy with the collections methods `isEmpty`/`nonEmpty`.

Comment (pathikrit) [inline core/src/main/scala/better/files/File.scala]: Good point. Done: https://github.com/pathikrit/better-files/commit/18c9cd51b7b2e503ff4944050ac5119470869e6e

**Comment by laughedelic** [comment]
@pathikrit could you publish a new release with this fix, please?

**Comment by pathikrit** [comment]
@laughedelic : In next week or two will release.

**Pull request merged**